Rescale sub int16 correctly

3l1 opened the pull request:
Summary:
we rescale to int32 using a different common scale(2*max(left scale,right scale))

the rescale(s) -> sub[int32] -> rescale from TOSA is converted to a single sub command stream instruction by Vela/Regor

Differential Revision: D83437623

digantdesai reviewed (approved):
Review automatically exported from Phabricator review in Meta.

digantdesai commented on backends/arm/test/ops/test_add.py:
```diff
@@ -276,9 +276,6 @@ def test_add_tensor_16a8w_tosa_INT(test_data: input_t1):
 
 @common.parametrize("test_data", Add.test_data)
 @common.XfailIfNoCorstone300
-@pytest.mark.xfail(
```
wrong PR?

digantdesai commented on backends/arm/tosa/quant_utils.py:
```diff
@@ -77,6 +77,59 @@ def insert_rescale_ops_to_int32_maxscale(
+def insert_rescale_ops_int16_to_int32_maxscale(
+    tosa_graph: Any, inputs: list[TosaArg], node: Node, tosa_spec=None
+) -> tuple[list[Any], float]:
+    """For ADD and SUB with int16 inputs, we rescale to int32 using a different common scale(2*max(left scale,right scale))
```
update this?

3l1 committed:
Rescale sub int16 correctly (#14650)

Summary:



we rescale to int32 using a different common scale(2*max(left scale,right scale))

the rescale(s) -> sub[int32] -> rescale from TOSA is converted to a single sub command stream instruction by Vela/Regor


bypass-github-export-checks
bypass-github-pytorch-ci-checks
bypass-github-executorch-ci-checks

Reviewed By: digantdesai

Differential Revision: D83437623
